sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -96,104 +96,77 @@ Below is a list of new compile-time constant properties supported with
 `annotated_ptr`.
 
 ```c++
-namespace sycl::ext::intel::experimental {
-
-enum class cache_level : /*unspecified*/ {
-    L1,
-    L2,
-    L3,
-    L4,
-};
-
-enum class cache_control_read_type : /* unspecified */ {
-    cached,
-    uncached,
-    streaming,
-    invalidate_after_read,
-    const_cached
+namespace sycl::ext {
+namespace intel::experimental {
+
+enum class cache_mode {
+  uncached,
+  cached,
+  streaming,
+  invalidate,
+  constant,
+  write_through,
+  write_back
 };
+using cache_mode = sycl::ext::oneapi::experimental::cache_mode;
+using cache_level = sycl::ext::oneapi::experimental::cache_level;
 
-enum class cache_control_write_type : /* unspecified */ {
-    uncached,
-    streaming,
-    write_through,
-    write_back
+struct read_hint_key {
+  template <typename... Cs>
+  using value_t = property_value<read_hint_key, Cs...>;
 };
 
-struct cache_control_read_key {
-  template<cache_control_read_type C, cache_level L>
-  using value_t = property_value<cache_control_read, C, std::integral_constant<cache_level, L>>;
+struct write_hint_key {
+  template <typename... Cs>
+  using value_t = property_value<write_hint_key, Cs...>;
 };
 
-struct cache_control_write_key {
-  template<cache_control_write_type C, cache_level L>
-  using value_t = property_value<cache_control_write, C, std::integral_constant<cache_level, L>>;
-};
-
-template<cache_control_read_type C, cache_level L>
-inline constexpr cache_control_read_key::value_t<C, L> cache_control_read;
-
-template<cache_control_write_type C, cache_level L>
-inline constexpr cache_control_write_key::value_t<C, L> cache_control_write;
-
-template<>
-struct is_property_key<cache_control_read_key> : std::true_type {};
+template <typename... Cs>
+inline constexpr read_hint_key::value_t<Cs...> read_hint;
 
-template<>
-struct is_property_key<cache_control_write_key> : std::true_type {};
+template <typename... Cs>
+inline constexpr write_hint_key::value_t<Cs...> write_hint;
 
-template<typename T, typename PropertyListT>
-struct is_property_key_of<
-  cache_control_read_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+} // namespace intel::experimental
 
-template<typename T, typename PropertyListT>
-struct is_property_key_of<
-  cache_control_write_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+using namespace intel::experimental;
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::cached, L>
-cache_control_read_cached;
+using read_hint_key = intel::experimental::read_hint_key;
```

Comment:
Yes, unneeded duplicate lines have been removed.

Reply:
This problem still exists.  I do not think we want this `using read_hint_key` because it pollutes the `sycl::ext::oneapi::experimental` namespace.  Instead, the trait can be specialized like this:

```
template <> struct is_property_key<intel::experimental::read_hint_key> : std::true_type {};
```

Make sure the implementation in the header does not also have this problem.